rest/codegen/server.go

```diff
@@ -1,11 +1,288 @@
 package codegen
 
 import (
+	"fmt"
+	"text/template"
+
 	"goa.design/goa.v2/codegen"
+	goadesign "goa.design/goa.v2/design"
 	"goa.design/goa.v2/rest/design"
+	"net/http"
 )
 
+//todo: in encode headers and mappings
+//todo: in decode payload, query and paths
+
+const serverT = `
+{{define "handlers_struct" -}}
+// {{.Name}}Handlers lists the {{.Name}} service endpoint HTTP handlers.
+type {{.Name}}Handlers struct {
+	{{- range .Actions}}
+	{{goify .Name true}} http.Handler
+	{{- end}}
+}
+{{end}}
+
+{{define "handlers_constructor" -}}
+// New{{.Name}}Handlers instantiates HTTP handlers for all the {{.Name}} service endpoints.
+func New{{.Name}}Handlers(
+	e *endpoints.{{.Name}},
+	dec rest.RequestDecoderFunc,
+	enc rest.ResponseEncoderFunc,
+	logger goa.Logger,
+) *{{.Name}}Handlers {
+	return &{{.Name}}Handlers{
+		{{- range .Actions}}
+		{{.Name}}: NewCreate{{$.Name}}Handler(e.{{.Name}}, dec, enc, logger),
+		{{- end}}
+	}
+}
+{{end}}
+
+{{define "mount_handlers" -}}
+// Mount{{.Name}}Handlers configures the mux to serve the {{.Name}} endpoints.
+func Mount{{.Name}}Handlers(mux rest.ServeMux, h *{{.Name}}Handlers) {
+	{{- range .Actions}}
+	Mount{{.Name}}{{$.Name}}Handler(mux, h.{{.Name}})
+	{{- end}}
+}
+{{end}}
+
+{{define "mount_action_handler" -}}
+// Mount{{.Name}}{{.Resource.Name}}Handler configures the mux to serve the
+// "{{.Resource.Name}}" service "{{.Name}}" endpoint.
+func Mount{{.Name}}{{.Resource.Name}}Handler(mux rest.ServeMux, h http.Handler) {
+	{{- range .Routes}}
+	mux.Handle("{{.Method}}", "{{.FullPath}}", h)
+	{{- end}}
+}
+{{end}}
+
+{{define "action_handler_constructor" -}}
+// New{{.Name}}{{.Resource.Name}}Handler creates a HTTP handler which loads the HTTP
+// request and calls the "{{.Resource.Name}}" service "{{.Name}}" endpoint.
+// The middleware is mounted so it executes after the request is loaded and thus
+// may access the request state via the rest package ContextXXX functions.
+func New{{.Name}}{{.Resource.Name}}Handler(
+	endpoint goa.Endpoint,
+	dec rest.RequestDecoderFunc,
+	enc rest.ResponseEncoderFunc,
+	logger goa.Logger,
+) http.Handler {
+	var (
+		decodeRequest  = {{.Name}}{{.Resource.Name}}DecodeRequest(dec)
```

Comment:
I do agree.

But what happens when i want to override the decoder with my own custom decoder and payload data. Than i should also override the constructor because the bindings are not there.